### ISSUE TITLE
Learned SDE is quite smooth

### ISSUE THREAD

**Issue by wangyixu14** [google-research/torchsde#113]
Hi,

I am trying to modify the latent_sde_lorenz example to learn an SDE from a simple 2-dimension noise data. The code works well but the only problem is that the learned SDE will always generate quite smooth samples. 
![image](https://user-images.githubusercontent.com/34372413/164505228-34e09567-9656-4e35-a778-ad9b9b9e2e60.png)

I'd like the learned SDE to show the same amount of stochasticity as the data, any idea how to deal with this? Is there any parameter that needs to be tuned?

**Comment by patrick-kidger** [comment]
This is a known problem with latent SDEs -- the diffusion is always a bit too small. I think this cpuld be mitigated by tweaking the training objective (probably the reconstruction loss encourages small diffusions), or by additionally training wrt a discriminator (see [Neural SDEs as Infinite-Dimensional GANs](https://arxiv.org/abs/2102.03657)). To my knowledge no-one has really tried this, or tried working out the details though.

**Comment by lxuechen** [comment]
Have you tried increasing the penalty on the path-space KL? 

The prior diffusion is unlikely to collapse, since it's forced to match multiple posterior diffusions approximately, simultaneously.

**Comment by wangyixu14** [comment]
> This is a known problem with latent SDEs -- the diffusion is always a bit too small. I think this could be mitigated by tweaking the training objective (probably the reconstruction loss encourages small diffusions), or by additionally training wrt a discriminator (see [Neural SDEs as Infinite-Dimensional GANs](https://arxiv.org/abs/2102.03657)). To my knowledge no-one has really tried this, or tried working out the details though.

Thanks for the reply. I am trying to adjust some loss functions. 

> Have you tried increasing the penalty on the path-space KL?
> 
> The prior diffusion is unlikely to collapse, since it's forced to match multiple posterior diffusions approximately, simultaneously.

Thank you, so which loss should I increase **log_pxs** or **logqp0 + logqp_path**?

**Comment by lxuechen** [comment]
I'm referring to the `logqp_path ` part.

**Comment by wangyixu14** [comment]
> I'm referring to the `logqp_path ` part.

Yes, it helps! Thank you guys for the timely reply. I am closing this issue.

**Comment by lxuechen** [comment]
> > I'm referring to the `logqp_path ` part.
> 
> Yes, it helps! Thank you guys for the timely reply. I am closing this issue.

Awesome! Feel free to reopen if/when you have follow-up questions.

**Comment by glatteis** [comment]
I investigated this issue during my master's thesis, as a side effect to what the topic was. I used an SDE that has two attractors with noise that makes it tip from one attractor to the other, called a zero-dimensional energy balance model:

![image](https://github.com/google-research/torchsde/assets/8374054/791a3dd2-992d-41fe-ac83-009432904d6c)

I measured the _tipping rate_ of the trained latent SDE and found out it is always too small compared to the data:

![image](https://github.com/google-research/torchsde/assets/8374054/50be0df7-16f7-4514-ba80-ba3b8cf30842)

Additionally, the latent SDE cannot match the _shape_ of the diffusion term:

![image](https://github.com/google-research/torchsde/assets/8374054/3508f013-07f4-4acd-b869-6855b1f56fe7)

Why is this? Reconsider the loss function:

![image](https://github.com/google-research/torchsde/assets/8374054/67c6638f-ebb7-4a26-8446-6f952c330453)

We are optimising a balance between the log-likelihood loss L_E and the KL loss L_KL. What does the balance look like for different values of beta? I varied the noise amplitude and plotted the scores for the final trained model:

![image](https://github.com/google-research/torchsde/assets/8374054/7b0f4bb9-cea7-4394-a6b4-a145d680fbb2)

So the data gets harder to approximate (higher=worse for KL and lower=worse for expectation) and the diffusion size grows linearly. If we use the trained latent SDE, for instance for beta=10, and we swap out the original diffusion for a constant diffusion with a pre-defined value, and then we plot this balance, we get this:

![image](https://github.com/google-research/torchsde/assets/8374054/c90d22ed-e535-42da-a080-c1628f50ea89)

The diffusion size is chosen at the point where these scores meet.

So the diffusion size of the latent SDE kinda related to the data noise level, but only in the sense that if the data becomes harder to predict, it gets larger because the point where the KL and log-likelihood loss meet moves further to the right.

**Comment by glatteis** [comment]
Hey folks, we finally published our paper on this problem (I commented the ideas this thread in 2023 and deleted them while our paper was in progress): https://pubs.aip.org/aip/cha/article/35/6/063139/3350739/Improving-the-noise-estimation-of-latent-neural
